GetStream/stream-chat-android
# [2969] Disable commands when text or attachments are present and disa…

### 🎯 Goal

Attachments should not be mixed with commands because the two are incompatible.
In order to signal that to the user we aim to remove the presence of commands suggestions and disable the commands buttom when attachments are present.

The user is also now unable to attach new content until they have dismissed the suggestions popup.

### 🛠 Implementation details

Introduced a new `SuggestionListControllerListener` to notify 'MessageInputView' when commands, mentions, and visibility changes occur.

Check for atttachments in `selectedAttachmentsCountListener` inside `MessageInputView` and enable/ disable command popups accordingly.

### 🎨 UI Changes

_Add relevant screenshots_

| Before | After |
| --- | --- |
| ![with_attachment_before](https://user-images.githubusercontent.com/37080097/152803914-55add65b-81fc-4da9-88d9-5b0654e0019a.png) | ![with_attachments_after](https://user-images.githubusercontent.com/37080097/152803984-50f5d11c-e226-4040-b041-13068bef9e7d.png) |
| ![with_commands_before](https://user-images.githubusercontent.com/37080097/152803921-1d6ead58-f4dc-4783-a51f-9e2303c5dafc.png) | ![with_commands_after](https://user-images.githubusercontent.com/37080097/152803995-53a3500a-a6b5-413a-b41c-3cc4f9ef4f2d.png) |
| ![with_mentions_before](https://user-images.githubusercontent.com/37080097/152803925-fb31a725-ffa3-41e5-b771-6cb046bff501.png) | ![with_mentions_after](https://user-images.githubusercontent.com/37080097/152804000-fd0f53ba-a83d-44c7-8122-3e874fb7497b.png) |

### 🧪 Testing

This is a permutation based improvement so please follow the given instructions and feel free to improvise scenarios beyond them as it is easy to miss an edge case here

Scenario one:

1. Select an attachments or attachment 
2. Type in "/"

Expected result: commands popup should not be visible 

Scenario Two:

1. Type in "/"

Expected result: Commands popup should be visible, attachment button disabled

Scenario Three:

1. Type in "@"

Expected result: Mentions popup should be visible, attachment button disabled

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![experience](https://user-images.githubusercontent.com/37080097/152801028-43cc4d5e-a445-49bb-a125-cbc88b730465.gif)
